diff -Naur FreeCAD-0.12.5284.orig/CMakeLists.txt FreeCAD-0.12.5284/CMakeLists.txt
--- FreeCAD-0.12.5284.orig/CMakeLists.txt	2012-04-08 07:23:26.474216557 -0500
+++ FreeCAD-0.12.5284/CMakeLists.txt	2012-04-08 19:14:13.606167248 -0500
@@ -85,6 +85,7 @@
 OPTION(FREECAD_BUILD_SANDBOX "Build the FreeCAD Sandbox module which is only for testing purposes" OFF)
 SET(RESOURCEDIR "${CMAKE_INSTALL_PREFIX}/data" CACHE PATH "Absolute path to the data directory.")
 SET(DOCDIR "" CACHE PATH "Absolute path to the documentation directory.")
+OPTION(USE_EXTERNAL_ZIPIOS "Use system installed zipios++ instead of the bundled." OFF)
 
 # ==============================================================================
 
diff -Naur FreeCAD-0.12.5284.orig/src/Base/CMakeLists.txt FreeCAD-0.12.5284/src/Base/CMakeLists.txt
--- FreeCAD-0.12.5284.orig/src/Base/CMakeLists.txt	2012-04-08 07:23:26.514215262 -0500
+++ FreeCAD-0.12.5284/src/Base/CMakeLists.txt	2012-04-08 19:12:39.956997353 -0500
@@ -269,7 +269,6 @@
 )
 
 SET(FreeCADBase_SRCS
-    ${zipios_SRCS}
     ${pycxx_SRCS}
     ${FreeCADBase_CPP_SRCS}
     ${FreeCADBase_HPP_SRCS}
@@ -281,6 +280,27 @@
     PreCompiled.h
 )
 
+# Use external zipios++ if specified.
+if(USE_EXTERNAL_ZIPIOS)
+    find_library(ZIPIOS_LIBRARY zipios)
+    find_path(ZIPIOS_INCLUDES zipios++/zipios-config.h)
+    if(ZIPIOS_LIBRARY)
+        message(STATUS "Found zipios++: ${ZIPIOS}")
+    endif()
+    if(ZIPIOS_INCLUDES)
+        message(STATUS "Found zipios++ headers.")
+    endif()
+    if(ZIPIOS_LIBRARY AND ZIPIOS_INCLUDES)
+        list(APPEND FreeCADBase_LIBS ${ZIPIOS_LIBRARY})
+        include_directories(${ZIPIOS_INCLUDES})
+    else()
+        message(FATAL_ERROR "Using external zipios++ was specified but was not found.")
+    endif()
+else(USE_EXTERNAL_ZIPIOS)
+    list(APPEND FreeCADBase_SRCS ${zipios_SRCS})
+endif(USE_EXTERNAL_ZIPIOS)
+
+
 if(MSVC)
 add_definitions(-D_PreComp_)
 ADD_MSVC_PRECOMPILED_HEADER("PreCompiled.h" "PreCompiled.cpp" FreeCADBase_CPP_SRCS)
