diff -Naur freecad-svn5184.orig/CMakeLists.txt freecad-svn5184.mod/CMakeLists.txt
--- freecad-svn5184.orig/CMakeLists.txt	2011-11-30 09:36:58.969275770 -0600
+++ freecad-svn5184.mod/CMakeLists.txt	2011-11-30 16:07:00.353807219 -0600
@@ -61,11 +61,12 @@
 if(WIN32)
     SET(CMAKE_INSTALL_PREFIX ${CMAKE_BINARY_DIR})
 else(WIN32)
-#    SET(CMAKE_INSTALL_PREFIX "/usr/lib/freecad")
+    SET(CMAKE_INSTALL_PREFIX "/usr/lib${LIB_SUFFIX}/freecad")
+    SET(INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
 endif(WIN32)
 
-# ================================================================================
-# == Win32 is default behaviour use the LibPack copied in Source tree ============
+# ==============================================================================
+# == Win32 is default behaviour use the LibPack copied in Source tree ==========
 if(MSVC)
     OPTION(FREECAD_LIBPACK_USE "Use the LibPack to Build FreeCAD (only Win32 so far)." ON)
     set(FREECAD_LIBPACK_DIR ${CMAKE_SOURCE_DIR} CACHE PATH  "Directory of the FreeCAD LibPack")
@@ -74,7 +75,7 @@
     set(FREECAD_LIBPACK_DIR ""  CACHE PATH  "Directory of the FreeCAD LibPack")
 endif(MSVC)
 
-# ================================================================================
+# ==============================================================================
 # == All the options for the build process    ============
 
 OPTION(FREECAD_BUILD_GUI "Build FreeCAD Gui. Otherwise you have only the command line and the Python import module." ON)
@@ -82,10 +83,10 @@
 OPTION(FREECAD_BUILD_CAM "Build the FreeCAD CAM module and the needed libs, be aware, unfinished code!" OFF)
 OPTION(FREECAD_BUILD_FEM "Build the FreeCAD FEM module, be aware, unfinished code!" ON)
 OPTION(FREECAD_BUILD_SANDBOX "Build the FreeCAD Sandbox module which is only for testing purposes" OFF)
+SET(RESOURCEDIR "${CMAKE_INSTALL_PREFIX}/data" CACHE PATH "Absolute path to the data directory.")
+SET(DOCDIR "" CACHE PATH "Absolute path to the documentation directory.")
 
-
-
-# ================================================================================
+# ==============================================================================
 
 if(FREECAD_LIBPACK_USE)
     # checking for a unique file in LibPack location to make sure the right version of the LibPack is there
diff -Naur freecad-svn5184.orig/data/examples/CMakeLists.txt freecad-svn5184.mod/data/examples/CMakeLists.txt
--- freecad-svn5184.orig/data/examples/CMakeLists.txt	2011-05-24 14:02:32.000000000 -0500
+++ freecad-svn5184.mod/data/examples/CMakeLists.txt	2011-11-30 11:56:01.050240955 -0600
@@ -12,7 +12,7 @@
 
 fc_copy_sources("data/examples" "Examples" ${Examples_Files})
 
-install(
+INSTALL(
     FILES
         ${Examples_Files}
     DESTINATION
diff -Naur freecad-svn5184.orig/src/App/CMakeLists.txt freecad-svn5184.mod/src/App/CMakeLists.txt
--- freecad-svn5184.orig/src/App/CMakeLists.txt	2011-06-14 09:11:52.000000000 -0500
+++ freecad-svn5184.mod/src/App/CMakeLists.txt	2011-11-30 16:08:36.464683016 -0600
@@ -3,6 +3,14 @@
     add_definitions(-DBOOST_DYN_LINK)
 endif(WIN32)
 
+IF(RESOURCEDIR)
+    add_definitions(-DRESOURCEDIR="${RESOURCEDIR}")
+ENDIF(RESOURCEDIR)
+
+IF(DOCDIR)
+    add_definitions(-DDOCDIR="${DOCDIR}")
+ENDIF(DOCDIR)
+
 include_directories(
     ${CMAKE_BINARY_DIR}/src
     ${CMAKE_SOURCE_DIR}/src
diff -Naur freecad-svn5184.orig/src/Base/CMakeLists.txt freecad-svn5184.mod/src/Base/CMakeLists.txt
--- freecad-svn5184.orig/src/Base/CMakeLists.txt	2011-01-26 05:56:38.000000000 -0600
+++ freecad-svn5184.mod/src/Base/CMakeLists.txt	2011-11-30 10:45:20.128198748 -0600
@@ -309,6 +309,9 @@
         LIBRARY DESTINATION lib
     )
 else(WIN32)
+    set_target_properties(FreeCADBase
+        PROPERTIES INSTALL_RPATH ${INSTALL_RPATH}
+    )
     INSTALL(TARGETS FreeCADBase
         LIBRARY DESTINATION lib
     )
diff -Naur freecad-svn5184.orig/src/Doc/CMakeLists.txt freecad-svn5184.mod/src/Doc/CMakeLists.txt
--- freecad-svn5184.orig/src/Doc/CMakeLists.txt	2011-01-28 08:38:53.000000000 -0600
+++ freecad-svn5184.mod/src/Doc/CMakeLists.txt	2011-11-30 10:38:04.225248987 -0600
@@ -1,5 +1,5 @@
 find_package(Doxygen)
-if( DOXYGEN_FOUND)
+if(DOXYGEN_FOUND)
 
     IF (DOXYGEN_DOT_EXECUTABLE)
         SET(HAVE_DOT YES)
diff -Naur freecad-svn5184.orig/src/Mod/Arch/CMakeLists.txt freecad-svn5184.mod/src/Mod/Arch/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Arch/CMakeLists.txt	2011-10-14 11:48:44.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Arch/CMakeLists.txt	2011-11-30 11:57:29.541317665 -0600
@@ -30,9 +30,7 @@
 
 fc_copy_sources("Mod/Arch" "Arch" ${all_files})
 
-install(
-    FILES
-        ${Arch_SRCS}
-    DESTINATION
-        Mod/Arch
+INSTALL(
+    FILES ${Arch_SRCS}
+    DESTINATION Mod/Arch
 )
diff -Naur freecad-svn5184.orig/src/Mod/Complete/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Complete/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Complete/App/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Complete/App/CMakeLists.txt	2011-11-30 11:00:47.662045120 -0600
@@ -38,6 +38,7 @@
 else(MSVC)
     set_target_properties(Complete PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Complete)
     set_target_properties(Complete PROPERTIES PREFIX "")
+    set_target_properties(Complete PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Complete DESTINATION lib)
+INSTALL(TARGETS Complete DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Complete/CMakeLists.txt freecad-svn5184.mod/src/Mod/Complete/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Complete/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Complete/CMakeLists.txt	2011-11-30 10:57:26.092627644 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Draft/CMakeLists.txt freecad-svn5184.mod/src/Mod/Draft/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Draft/CMakeLists.txt	2011-02-02 09:56:18.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Draft/CMakeLists.txt	2011-11-30 11:03:20.932041683 -0600
@@ -34,14 +34,14 @@
 
 fc_copy_sources("Mod/Draft" "Draft" ${all_files})
 
-install(
+
+INSTALL(
     FILES
         ${DraftLibs_SRCS}
     DESTINATION
         Mod/Draft/draftlibs
 )
-
-install(
+INSTALL(
     FILES
         ${Draft_SRCS}
     DESTINATION
diff -Naur freecad-svn5184.orig/src/Mod/Drawing/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Drawing/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Drawing/App/CMakeLists.txt	2011-11-17 04:31:04.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Drawing/App/CMakeLists.txt	2011-11-30 12:03:31.686356774 -0600
@@ -85,6 +85,7 @@
 else(MSVC)
     set_target_properties(Drawing PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Drawing)
     set_target_properties(Drawing PROPERTIES PREFIX "")
+    set_target_properties(Drawing PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Drawing DESTINATION lib)
+INSTALL(TARGETS Drawing DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Drawing/CMakeLists.txt freecad-svn5184.mod/src/Mod/Drawing/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Drawing/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Drawing/CMakeLists.txt	2011-11-30 11:03:56.361884884 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Drawing/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Drawing/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Drawing/Gui/CMakeLists.txt	2011-06-15 09:24:33.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Drawing/Gui/CMakeLists.txt	2011-11-30 11:05:11.758423159 -0600
@@ -83,6 +83,7 @@
 else(MSVC)
     set_target_properties(DrawingGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Drawing)
     set_target_properties(DrawingGui PROPERTIES PREFIX "")
+    set_target_properties(DrawingGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS DrawingGui DESTINATION lib)
+INSTALL(TARGETS DrawingGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Fem/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Fem/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Fem/App/CMakeLists.txt	2011-03-31 10:38:57.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Fem/App/CMakeLists.txt	2011-11-30 11:24:22.670819700 -0600
@@ -79,6 +79,7 @@
 else(MSVC)
     set_target_properties(Fem PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Fem)
     set_target_properties(Fem PROPERTIES PREFIX "")
+    set_target_properties(Fem PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Fem DESTINATION lib)
+INSTALL(TARGETS Fem DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Fem/CMakeLists.txt freecad-svn5184.mod/src/Mod/Fem/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Fem/CMakeLists.txt	2010-11-23 07:25:34.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Fem/CMakeLists.txt	2011-11-30 11:23:00.206515462 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Fem/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Fem/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Fem/Gui/CMakeLists.txt	2010-10-24 05:36:10.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Fem/Gui/CMakeLists.txt	2011-11-30 11:23:44.905054292 -0600
@@ -66,6 +66,7 @@
 else(MSVC)
     set_target_properties(FemGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Fem)
     set_target_properties(FemGui PROPERTIES PREFIX "")
+    set_target_properties(FemGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS FemGui DESTINATION lib)
+INSTALL(TARGETS FemGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Idf/CMakeLists.txt freecad-svn5184.mod/src/Mod/Idf/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Idf/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Idf/CMakeLists.txt	2011-11-30 11:25:49.758972604 -0600
@@ -48,23 +48,21 @@
 
 fc_copy_sources("Mod/Idf" "Idf" ${all_files})
 
-install(
+INSTALL(
     FILES
         ${IdfLibs_SRCS}
     DESTINATION
         Mod/Idf/Idflibs
 )
-
-install(
+INSTALL(
     FILES
         ${IdfLang_SRCS}
     DESTINATION
         Mod/Idf/Languages
 )
-
-install(
+INSTALL(
     FILES
         ${Idf_SRCS}
     DESTINATION
         Mod/Idf
- )
+)
diff -Naur freecad-svn5184.orig/src/Mod/Image/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Image/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Image/App/CMakeLists.txt	2011-01-25 09:40:19.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Image/App/CMakeLists.txt	2011-11-30 11:27:17.458105233 -0600
@@ -44,6 +44,7 @@
 else(MSVC)
     set_target_properties(Image PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Image)
     set_target_properties(Image PROPERTIES PREFIX "")
+    set_target_properties(Image PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Image DESTINATION lib)
+INSTALL(TARGETS Image DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Image/CMakeLists.txt freecad-svn5184.mod/src/Mod/Image/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Image/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Image/CMakeLists.txt	2011-11-30 11:26:36.060458740 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Image/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Image/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Image/Gui/CMakeLists.txt	2011-01-24 12:06:55.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Image/Gui/CMakeLists.txt	2011-11-30 11:27:58.559761333 -0600
@@ -69,6 +69,7 @@
 else(MSVC)
     set_target_properties(ImageGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Image)
     set_target_properties(ImageGui PROPERTIES PREFIX "")
+    set_target_properties(ImageGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS ImageGui DESTINATION lib)
+INSTALL(TARGETS ImageGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Import/CMakeLists.txt freecad-svn5184.mod/src/Mod/Import/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Import/CMakeLists.txt	2011-01-04 06:35:19.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Import/CMakeLists.txt	2011-11-30 11:28:45.368230781 -0600
@@ -4,10 +4,10 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
     DESTINATION
         Mod/Import
-)
+)   
diff -Naur freecad-svn5184.orig/src/Mod/Import/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Import/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Import/Gui/CMakeLists.txt	2011-02-02 06:29:29.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Import/Gui/CMakeLists.txt	2011-11-30 11:29:14.799268344 -0600
@@ -56,6 +56,7 @@
 else(MSVC)
     set_target_properties(ImportGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Import)
     set_target_properties(ImportGui PROPERTIES PREFIX "")
+    set_target_properties(ImportGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS ImportGui DESTINATION lib)
+INSTALL(TARGETS ImportGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Inspection/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Inspection/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Inspection/App/CMakeLists.txt	2011-02-02 06:29:29.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Inspection/App/CMakeLists.txt	2011-11-30 11:30:36.267604260 -0600
@@ -50,6 +50,7 @@
 else(MSVC)
     set_target_properties(Inspection PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Inspection)
     set_target_properties(Inspection PROPERTIES PREFIX "")
+    set_target_properties(Inspection PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
-
-install(TARGETS Inspection DESTINATION lib)
+
+INSTALL(TARGETS Inspection DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Inspection/CMakeLists.txt freecad-svn5184.mod/src/Mod/Inspection/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Inspection/CMakeLists.txt	2011-01-26 03:20:55.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Inspection/CMakeLists.txt	2011-11-30 11:29:53.774993917 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Inspection/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Inspection/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Inspection/Gui/CMakeLists.txt	2011-02-02 06:29:29.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Inspection/Gui/CMakeLists.txt	2011-11-30 11:31:46.555305748 -0600
@@ -47,26 +47,27 @@
     ViewProviderInspection.cpp
     ViewProviderInspection.h
     Workbench.cpp
-    Workbench.h
-)
-
-add_library(InspectionGui SHARED ${InspectionGui_SRCS})
-target_link_libraries(InspectionGui ${InspectionGui_LIBS})
-fc_copy_script("Mod/Inspection" "InspectionGui" InitGui.py)
-
-if(MSVC)
-    set_target_properties(InspectionGui PROPERTIES SUFFIX ".pyd")
-    set_target_properties(InspectionGui PROPERTIES DEBUG_OUTPUT_NAME "InspectionGui_d")
-    set_target_properties(InspectionGui PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Inspection)
-    set_target_properties(InspectionGui PROPERTIES PREFIX "../")
-elseif(MINGW)
-    set_target_properties(InspectionGui PROPERTIES SUFFIX ".pyd")
-    set_target_properties(InspectionGui PROPERTIES DEBUG_OUTPUT_NAME "InspectionGui_d")
-    set_target_properties(InspectionGui PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Inspection)
-    set_target_properties(InspectionGui PROPERTIES PREFIX "")
-else(MSVC)
-    set_target_properties(InspectionGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Inspection)
-    set_target_properties(InspectionGui PROPERTIES PREFIX "")
-endif(MSVC)
-
-install(TARGETS InspectionGui DESTINATION lib)
+    Workbench.h
+)
+
+add_library(InspectionGui SHARED ${InspectionGui_SRCS})
+target_link_libraries(InspectionGui ${InspectionGui_LIBS})
+fc_copy_script("Mod/Inspection" "InspectionGui" InitGui.py)
+
+if(MSVC)
+    set_target_properties(InspectionGui PROPERTIES SUFFIX ".pyd")
+    set_target_properties(InspectionGui PROPERTIES DEBUG_OUTPUT_NAME "InspectionGui_d")
+    set_target_properties(InspectionGui PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Inspection)
+    set_target_properties(InspectionGui PROPERTIES PREFIX "../")
+elseif(MINGW)
+    set_target_properties(InspectionGui PROPERTIES SUFFIX ".pyd")
+    set_target_properties(InspectionGui PROPERTIES DEBUG_OUTPUT_NAME "InspectionGui_d")
+    set_target_properties(InspectionGui PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Inspection)
+    set_target_properties(InspectionGui PROPERTIES PREFIX "")
+else(MSVC)
+    set_target_properties(InspectionGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Inspection)
+    set_target_properties(InspectionGui PROPERTIES PREFIX "")
+    set_target_properties(InspectionGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
+endif(MSVC)
+
+INSTALL(TARGETS InspectionGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Mesh/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Mesh/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Mesh/App/CMakeLists.txt	2011-05-20 09:39:48.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Mesh/App/CMakeLists.txt	2011-11-30 11:33:47.854316080 -0600
@@ -347,6 +347,7 @@
 else(MSVC)
     set_target_properties(Mesh PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Mesh)
     set_target_properties(Mesh PROPERTIES PREFIX "")
+    set_target_properties(Mesh PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Mesh DESTINATION lib)
+INSTALL(TARGETS Mesh DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Mesh/CMakeLists.txt freecad-svn5184.mod/src/Mod/Mesh/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Mesh/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Mesh/CMakeLists.txt	2011-11-30 11:33:19.819243657 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Mesh/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Mesh/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Mesh/Gui/CMakeLists.txt	2010-11-05 09:59:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Mesh/Gui/CMakeLists.txt	2011-11-30 11:34:26.387041214 -0600
@@ -124,6 +124,7 @@
 else(MSVC)
     set_target_properties(MeshGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Mesh)
     set_target_properties(MeshGui PROPERTIES PREFIX "")
+    set_target_properties(MeshGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS MeshGui DESTINATION lib)
+INSTALL(TARGETS MeshGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/MeshPart/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/MeshPart/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/MeshPart/App/CMakeLists.txt	2010-10-24 05:36:10.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/MeshPart/App/CMakeLists.txt	2011-11-30 11:35:26.526051628 -0600
@@ -66,6 +66,7 @@
 else(MSVC)
     set_target_properties(MeshPart PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/MeshPart)
     set_target_properties(MeshPart PROPERTIES PREFIX "")
+    set_target_properties(MeshPart PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS MeshPart DESTINATION lib)
+INSTALL(TARGETS MeshPart DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/MeshPart/CMakeLists.txt freecad-svn5184.mod/src/Mod/MeshPart/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/MeshPart/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/MeshPart/CMakeLists.txt	2011-11-30 11:34:57.482012457 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/MeshPart/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/MeshPart/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/MeshPart/Gui/CMakeLists.txt	2010-10-02 18:53:11.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/MeshPart/Gui/CMakeLists.txt	2011-11-30 11:35:53.700152675 -0600
@@ -69,6 +69,7 @@
 else(MSVC)
     set_target_properties(MeshPartGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/MeshPart)
     set_target_properties(MeshPartGui PROPERTIES PREFIX "")
+    set_target_properties(MeshPartGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS MeshPartGui DESTINATION lib)
+INSTALL(TARGETS MeshPartGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Part/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Part/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Part/App/CMakeLists.txt	2011-11-17 04:31:04.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Part/App/CMakeLists.txt	2011-11-30 11:36:57.685036101 -0600
@@ -252,6 +252,7 @@
 else(MSVC)
     set_target_properties(Part PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Part)
     set_target_properties(Part PROPERTIES PREFIX "")
+    set_target_properties(Part PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Part DESTINATION lib)
+INSTALL(TARGETS Part DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Part/CMakeLists.txt freecad-svn5184.mod/src/Mod/Part/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Part/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Part/CMakeLists.txt	2011-11-30 11:36:19.927285066 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Part/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Part/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Part/Gui/CMakeLists.txt	2011-10-15 08:31:58.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Part/Gui/CMakeLists.txt	2011-11-30 11:37:36.752743837 -0600
@@ -178,6 +178,7 @@
 else(MSVC)
     set_target_properties(PartGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Part)
     set_target_properties(PartGui PROPERTIES PREFIX "")
+    set_target_properties(PartGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS PartGui DESTINATION lib)
+INSTALL(TARGETS PartGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/PartDesign/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/PartDesign/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/PartDesign/App/CMakeLists.txt	2011-11-22 04:37:02.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/PartDesign/App/CMakeLists.txt	2011-11-30 11:38:57.340078457 -0600
@@ -105,6 +105,7 @@
 else(MSVC)
     set_target_properties(PartDesign PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/PartDesign)
     set_target_properties(PartDesign PROPERTIES PREFIX "")
+    set_target_properties(PartDesign PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS PartDesign DESTINATION lib)
+INSTALL(TARGETS PartDesign DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/PartDesign/CMakeLists.txt freecad-svn5184.mod/src/Mod/PartDesign/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/PartDesign/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/PartDesign/CMakeLists.txt	2011-11-30 11:38:26.413101317 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/PartDesign/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/PartDesign/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/PartDesign/Gui/CMakeLists.txt	2011-11-17 03:29:05.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/PartDesign/Gui/CMakeLists.txt	2011-11-30 11:39:37.893737252 -0600
@@ -134,6 +134,7 @@
 else(MSVC)
     set_target_properties(PartDesignGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/PartDesign)
     set_target_properties(PartDesignGui PROPERTIES PREFIX "")
+    set_target_properties(PartDesignGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS PartDesignGui DESTINATION lib)
+INSTALL(TARGETS PartDesignGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Points/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Points/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Points/App/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Points/App/CMakeLists.txt	2011-11-30 11:41:17.777434173 -0600
@@ -59,6 +59,7 @@
 else(MSVC)
     set_target_properties(Points PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Points)
     set_target_properties(Points PROPERTIES PREFIX "")
+    set_target_properties(Points PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Points DESTINATION lib)
+INSTALL(TARGETS Points DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Points/CMakeLists.txt freecad-svn5184.mod/src/Mod/Points/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Points/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Points/CMakeLists.txt	2011-11-30 11:40:18.488394768 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Points/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Points/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Points/Gui/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Points/Gui/CMakeLists.txt	2011-11-30 11:42:01.941973792 -0600
@@ -69,6 +69,7 @@
 else(MSVC)
     set_target_properties(PointsGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Points)
     set_target_properties(PointsGui PROPERTIES PREFIX "")
+    set_target_properties(PointsGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS PointsGui DESTINATION lib)
+INSTALL(TARGETS PointsGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Raytracing/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Raytracing/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Raytracing/App/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Raytracing/App/CMakeLists.txt	2011-11-30 11:42:57.676130964 -0600
@@ -74,6 +74,7 @@
 else(MSVC)
     set_target_properties(Raytracing PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Raytracing)
     set_target_properties(Raytracing PROPERTIES PREFIX "")
+    set_target_properties(Raytracing PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Raytracing DESTINATION lib)
+INSTALL(TARGETS Raytracing DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Raytracing/CMakeLists.txt freecad-svn5184.mod/src/Mod/Raytracing/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Raytracing/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Raytracing/CMakeLists.txt	2011-11-30 11:42:29.349067575 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Raytracing/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Raytracing/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Raytracing/Gui/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Raytracing/Gui/CMakeLists.txt	2011-11-30 11:43:26.181188497 -0600
@@ -89,6 +89,7 @@
 else(MSVC)
     set_target_properties(RaytracingGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Raytracing)
     set_target_properties(RaytracingGui PROPERTIES PREFIX "")
+    set_target_properties(RaytracingGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS RaytracingGui DESTINATION lib)
+INSTALL(TARGETS RaytracingGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/ReverseEngineering/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/ReverseEngineering/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/ReverseEngineering/App/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/ReverseEngineering/App/CMakeLists.txt	2011-11-30 11:44:24.234269226 -0600
@@ -47,6 +47,7 @@
 else(MSVC)
     set_target_properties(ReverseEngineering PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/ReverseEngineering)
     set_target_properties(ReverseEngineering PROPERTIES PREFIX "")
+    set_target_properties(ReverseEngineering PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS ReverseEngineering DESTINATION lib)
+INSTALL(TARGETS ReverseEngineering DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/ReverseEngineering/CMakeLists.txt freecad-svn5184.mod/src/Mod/ReverseEngineering/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/ReverseEngineering/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/ReverseEngineering/CMakeLists.txt	2011-11-30 11:43:50.583381744 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Robot/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Robot/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Robot/App/CMakeLists.txt	2011-07-26 08:37:24.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Robot/App/CMakeLists.txt	2011-11-30 11:45:31.977029697 -0600
@@ -137,6 +137,7 @@
 else(MSVC)
     set_target_properties(Robot PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Robot)
     set_target_properties(Robot PROPERTIES PREFIX "")
+    set_target_properties(Robot PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Robot DESTINATION lib)
+INSTALL(TARGETS Robot DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Robot/CMakeLists.txt freecad-svn5184.mod/src/Mod/Robot/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Robot/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Robot/CMakeLists.txt	2011-11-30 11:45:04.496938137 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Robot/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Robot/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Robot/Gui/CMakeLists.txt	2011-07-26 08:37:24.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Robot/Gui/CMakeLists.txt	2011-11-30 11:46:01.680047793 -0600
@@ -162,6 +162,7 @@
 else(MSVC)
     set_target_properties(RobotGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Robot)
     set_target_properties(RobotGui PROPERTIES PREFIX "")
+    set_target_properties(RobotGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS RobotGui DESTINATION lib)
+INSTALL(TARGETS RobotGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Sandbox/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Sandbox/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Sandbox/App/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Sandbox/App/CMakeLists.txt	2011-11-30 11:47:17.199551470 -0600
@@ -46,6 +46,7 @@
 else(MSVC)
     set_target_properties(Sandbox PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Sandbox)
     set_target_properties(Sandbox PROPERTIES PREFIX "")
+    set_target_properties(Sandbox PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Sandbox DESTINATION lib)
+INSTALL(TARGETS Sandbox DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Sandbox/CMakeLists.txt freecad-svn5184.mod/src/Mod/Sandbox/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Sandbox/CMakeLists.txt	2010-08-24 06:42:52.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Sandbox/CMakeLists.txt	2011-11-30 11:46:39.023813388 -0600
@@ -4,10 +4,10 @@
 	add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
-	FILES
-		Init.py
-		InitGui.py
-	DESTINATION
-		Mod/Sandbox
+INSTALL(
+    FILES
+        Init.py
+        InitGui.py
+    DESTINATION
+        Mod/Sandbox
 )
diff -Naur freecad-svn5184.orig/src/Mod/Sandbox/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Sandbox/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Sandbox/Gui/CMakeLists.txt	2011-07-08 08:07:49.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Sandbox/Gui/CMakeLists.txt	2011-11-30 11:47:59.296160060 -0600
@@ -57,6 +57,7 @@
 else(MSVC)
     set_target_properties(SandboxGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Sandbox)
     set_target_properties(SandboxGui PROPERTIES PREFIX "")
+    set_target_properties(SandboxGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS SandboxGui DESTINATION lib)
+INSTALL(TARGETS SandboxGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Sketcher/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Sketcher/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Sketcher/App/CMakeLists.txt	2011-09-07 09:40:01.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Sketcher/App/CMakeLists.txt	2011-11-30 11:48:49.511500367 -0600
@@ -121,6 +121,7 @@
 else(MSVC)
     set_target_properties(Sketcher PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Sketcher)
     set_target_properties(Sketcher PROPERTIES PREFIX "")
+    set_target_properties(Sketcher PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Sketcher DESTINATION lib)
+INSTALL(TARGETS Sketcher DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Sketcher/CMakeLists.txt freecad-svn5184.mod/src/Mod/Sketcher/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Sketcher/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Sketcher/CMakeLists.txt	2011-11-30 11:48:21.240434767 -0600
@@ -4,7 +4,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Sketcher/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Sketcher/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Sketcher/Gui/CMakeLists.txt	2011-10-03 17:57:56.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Sketcher/Gui/CMakeLists.txt	2011-11-30 11:49:16.609604780 -0600
@@ -113,6 +113,7 @@
 else(MSVC)
     set_target_properties(SketcherGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Sketcher)
     set_target_properties(SketcherGui PROPERTIES PREFIX "")
+    set_target_properties(SketcherGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS SketcherGui DESTINATION lib)
+INSTALL(TARGETS SketcherGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Start/App/CMakeLists.txt freecad-svn5184.mod/src/Mod/Start/App/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Start/App/CMakeLists.txt	2011-05-16 11:06:57.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Start/App/CMakeLists.txt	2011-11-30 11:50:12.614753899 -0600
@@ -37,6 +37,7 @@
 else(MSVC)
     set_target_properties(Start PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Start)
     set_target_properties(Start PROPERTIES PREFIX "")
+    set_target_properties(Start PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS Start DESTINATION lib)
+INSTALL(TARGETS Start DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Start/CMakeLists.txt freecad-svn5184.mod/src/Mod/Start/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Start/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Start/CMakeLists.txt	2011-11-30 19:08:17.276035136 -0600
@@ -2,9 +2,10 @@
 add_subdirectory(App)
 if(FREECAD_BUILD_GUI)
     add_subdirectory(Gui)
+    add_subdirectory(StartPage)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Start/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Start/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Start/Gui/CMakeLists.txt	2011-09-13 09:34:13.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Start/Gui/CMakeLists.txt	2011-11-30 11:50:41.919785412 -0600
@@ -70,6 +70,7 @@
 else(MSVC)
     set_target_properties(StartGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Start)
     set_target_properties(StartGui PROPERTIES PREFIX "")
+    set_target_properties(StartGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS StartGui DESTINATION lib)
+INSTALL(TARGETS StartGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/TemplatePyMod/CMakeLists.txt freecad-svn5184.mod/src/Mod/TemplatePyMod/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/TemplatePyMod/CMakeLists.txt	2011-08-08 14:17:10.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/TemplatePyMod/CMakeLists.txt	2011-11-30 11:51:17.723602252 -0600
@@ -9,13 +9,14 @@
 	DocumentObject.py
 )
 
-install(
-		 FILES
-		 		 Init.py
-		 		 InitGui.py
-		 		 Commands.py
-				 FeaturePython.py
-		 		 PythonQt.py
-		 		 Tests.py
-		 DESTINATION
-		 		 Mod/TemplatePyMod)
+INSTALL(
+    FILES
+        Init.py
+        InitGui.py
+        Commands.py
+        FeaturePython.py
+        PythonQt.py
+        Tests.py
+    DESTINATION
+        Mod/TemplatePyMod
+)
diff -Naur freecad-svn5184.orig/src/Mod/Test/CMakeLists.txt freecad-svn5184.mod/src/Mod/Test/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Test/CMakeLists.txt	2011-07-13 05:10:30.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Test/CMakeLists.txt	2011-11-30 11:51:58.554253017 -0600
@@ -20,7 +20,7 @@
 
 fc_copy_sources("Mod/Test" "Test" ${Test_SRCS})
 
-install(
+INSTALL(
     FILES
         ${Test_SRCS}
     DESTINATION
diff -Naur freecad-svn5184.orig/src/Mod/Test/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Test/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Test/Gui/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Test/Gui/CMakeLists.txt	2011-11-30 11:52:37.617962226 -0600
@@ -65,6 +65,7 @@
 else(MSVC)
     set_target_properties(QtUnitGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Test)
     set_target_properties(QtUnitGui PROPERTIES PREFIX "")
+    set_target_properties(QtUnitGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS QtUnitGui DESTINATION lib)
+INSTALL(TARGETS QtUnitGui DESTINATION lib)
diff -Naur freecad-svn5184.orig/src/Mod/Web/CMakeLists.txt freecad-svn5184.mod/src/Mod/Web/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Web/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Web/CMakeLists.txt	2011-11-30 11:53:02.772131074 -0600
@@ -3,7 +3,7 @@
     add_subdirectory(Gui)
 endif(FREECAD_BUILD_GUI)
 
-install(
+INSTALL(
     FILES
         Init.py
         InitGui.py
diff -Naur freecad-svn5184.orig/src/Mod/Web/Gui/CMakeLists.txt freecad-svn5184.mod/src/Mod/Web/Gui/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Web/Gui/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5184.mod/src/Mod/Web/Gui/CMakeLists.txt	2011-11-30 11:54:00.295230454 -0600
@@ -52,6 +52,7 @@
 else(MSVC)
     set_target_properties(WebGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Web)
     set_target_properties(WebGui PROPERTIES PREFIX "")
+    set_target_properties(WebGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
-install(TARGETS WebGui DESTINATION lib)
+INSTALL(TARGETS WebGui DESTINATION lib)
diff -Naur freecad-svn5220.orig/src/Mod/Complete/Gui/CMakeLists.txt freecad-svn5220/src/Mod/Complete/Gui/CMakeLists.txt
--- freecad-svn5220.orig/src/Mod/Complete/Gui/CMakeLists.txt	2010-09-10 17:47:15.000000000 -0500
+++ freecad-svn5220/src/Mod/Complete/Gui/CMakeLists.txt	2011-12-05 10:20:10.830553148 -0600
@@ -44,6 +44,7 @@
 else(MSVC)
     set_target_properties(CompleteGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/Complete)
     set_target_properties(CompleteGui PROPERTIES PREFIX "")
+    set_target_properties(CompleteGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
 install(TARGETS CompleteGui DESTINATION lib)
diff -Naur freecad-svn5220.orig/src/Mod/ReverseEngineering/Gui/CMakeLists.txt freecad-svn5220/src/Mod/ReverseEngineering/Gui/CMakeLists.txt
--- freecad-svn5220.orig/src/Mod/ReverseEngineering/Gui/CMakeLists.txt	2010-09-11 04:58:50.000000000 -0500
+++ freecad-svn5220/src/Mod/ReverseEngineering/Gui/CMakeLists.txt	2011-12-05 10:50:21.815925156 -0600
@@ -54,6 +54,9 @@
 else(MSVC)
     set_target_properties(ReverseEngineeringGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Mod/ReverseEngineering)
     set_target_properties(ReverseEngineeringGui PROPERTIES PREFIX "")
+    set_target_properties(ReverseEngineeringGui
+                          PROPERTIES INSTALL_RPATH ${INSTALL_RPATH}
+    )
 endif(MSVC)
 
 install(TARGETS ReverseEngineeringGui DESTINATION lib)

diff -Naur freecad-svn5220.orig/src/App/CMakeLists.txt freecad-svn5220/src/App/CMakeLists.txt
--- freecad-svn5220.orig/src/App/CMakeLists.txt	2011-12-05 13:43:58.164206727 -0600
+++ freecad-svn5220/src/App/CMakeLists.txt	2011-12-05 13:35:16.637348028 -0600
@@ -184,6 +184,7 @@
     set_target_properties(FreeCADApp PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 else(MSVC)
     set_target_properties(FreeCADApp PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+    set_target_properties(FreeCADApp PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
 if(WIN32)
@@ -192,7 +193,7 @@
         LIBRARY DESTINATION lib
     )
 else(WIN32)
-    INSTALL(TARGETS  FreeCADApp
+    INSTALL(TARGETS FreeCADApp
         LIBRARY DESTINATION lib
     )
 endif(WIN32)
diff -Naur freecad-svn5220.orig/src/Base/CMakeLists.txt freecad-svn5220/src/Base/CMakeLists.txt
--- freecad-svn5220.orig/src/Base/CMakeLists.txt	2011-12-05 13:43:58.164206727 -0600
+++ freecad-svn5220/src/Base/CMakeLists.txt	2011-12-05 13:37:27.551041426 -0600
@@ -301,6 +301,7 @@
     set_target_properties(FreeCADBase PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 else(MSVC)
     set_target_properties(FreeCADBase PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+    set_target_properties(FreeCADBase PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
 if(WIN32)
@@ -309,9 +310,6 @@
         LIBRARY DESTINATION lib
     )
 else(WIN32)
-    set_target_properties(FreeCADBase
-        PROPERTIES INSTALL_RPATH ${INSTALL_RPATH}
-    )
     INSTALL(TARGETS FreeCADBase
         LIBRARY DESTINATION lib
     )
diff -Naur freecad-svn5220.orig/src/Gui/CMakeLists.txt freecad-svn5220/src/Gui/CMakeLists.txt
--- freecad-svn5220.orig/src/Gui/CMakeLists.txt	2011-07-25 09:59:26.000000000 -0500
+++ freecad-svn5220/src/Gui/CMakeLists.txt	2011-12-05 13:35:57.316009531 -0600
@@ -830,6 +830,7 @@
     set_target_properties(FreeCADGui PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 else(MSVC)
     set_target_properties(FreeCADGui PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+    set_target_properties(FreeCADGui PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
 if(WIN32)
diff -Naur freecad-svn5220.orig/src/Main/CMakeLists.txt freecad-svn5220/src/Main/CMakeLists.txt
--- freecad-svn5220.orig/src/Main/CMakeLists.txt	2011-08-07 06:09:01.000000000 -0500
+++ freecad-svn5220/src/Main/CMakeLists.txt	2011-12-05 13:42:07.577838112 -0600
@@ -59,6 +59,7 @@
     else(MSVC)
         set_target_properties(FreeCADMain PROPERTIES OUTPUT_NAME "FreeCAD")
         set_target_properties(FreeCADMain PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+        set_target_properties(FreeCADMain PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
     endif(MSVC)
 
     if(WIN32)
@@ -99,6 +100,7 @@
 else(MSVC)
     set_target_properties(FreeCADMainCmd PROPERTIES OUTPUT_NAME "FreeCADCmd")
     set_target_properties(FreeCADMainCmd PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+    set_target_properties(FreeCADMainCmd PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
 if(WIN32)
@@ -140,6 +142,7 @@
     set_target_properties(FreeCADMainPy PROPERTIES OUTPUT_NAME "FreeCAD")
     set_target_properties(FreeCADMainPy PROPERTIES PREFIX "")
     set_target_properties(FreeCADMainPy PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+    set_target_properties(FreeCADMainPy PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 endif(MSVC)
 
 if(WIN32)
@@ -178,10 +181,12 @@
 		set_target_properties(FreeCADGuiPy PROPERTIES SUFFIX ".pyd")
 		set_target_properties(FreeCADGuiPy PROPERTIES PREFIX "")
 		set_target_properties(FreeCADGuiPy PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+                set_target_properties(FreeCADGuiPy PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 	else(MSVC)
 		set_target_properties(FreeCADGuiPy PROPERTIES OUTPUT_NAME "FreeCADGui")
 		set_target_properties(FreeCADGuiPy PROPERTIES PREFIX "")
 		set_target_properties(FreeCADGuiPy PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+                set_target_properties(FreeCADGuiPy PROPERTIES INSTALL_RPATH ${INSTALL_RPATH})
 	endif(MSVC)
 
 	if(WIN32)
@@ -194,4 +199,4 @@
 			LIBRARY DESTINATION lib
 		)
 	endif(WIN32)
-endif(FREECAD_BUILD_GUI)
\ No newline at end of file
+endif(FREECAD_BUILD_GUI)
