diff -Naur freecad-0.12.orig/src/Mod/Image/Gui/GLImageBox.cpp freecad-0.12/src/Mod/Image/Gui/GLImageBox.cpp
--- freecad-0.12.orig/src/Mod/Image/Gui/GLImageBox.cpp	2010-02-18 05:03:53.000000000 -0600
+++ freecad-0.12/src/Mod/Image/Gui/GLImageBox.cpp	2012-01-01 16:49:09.041859287 -0600
@@ -27,6 +27,7 @@
 # include <GL/glext.h>
 #endif
 
+#include <GL/glu.h>
 #include "GLImageBox.h"
 
 using namespace ImageGui;
