diff -Naur freecad-svn5184.orig/src/Mod/Start/StartPage/CMakeLists.txt freecad-svn5184.mod/src/Mod/Start/StartPage/CMakeLists.txt
--- freecad-svn5184.orig/src/Mod/Start/StartPage/CMakeLists.txt	1969-12-31 18:00:00.000000000 -0600
+++ freecad-svn5184.mod/src/Mod/Start/StartPage/CMakeLists.txt	2011-11-30 19:07:40.962208203 -0600
@@ -0,0 +1,31 @@
+SET(StartPage_SRCS
+	StartPage.py
+	__init__.py
+)
+
+SET(StartPage_DATA 
+	ArchDesign.py
+	DefaultWorkbench.py
+	LoadDrawingExample.py
+	LoadPartDesignExample.py
+	LoadRobotExample.py
+	LoadSchenkel.py
+	LoadMRU0.py
+	LoadMRU1.py
+	LoadMRU2.py
+	Mesh.py
+	PartDesign.py
+	Background.jpg
+	FreeCAD.png
+	PartDesign.png
+	ArchDesign.png
+	Mesh.png
+	Complete.png
+	PartDesignExample.png
+	ArchExample.png
+	web.png
+)
+
+INSTALL(FILES ${StartPage_SRCS} ${StartPage_DATA}
+	DESTINATION Mod/Start/StartPage
+)
